scipy/stats/_covariance.py

```diff
@@ -89,6 +90,34 @@ def from_diagonal(diagonal):
         does not lie in the span of the columns of the covariance matrix. The
         convention taken here is to treat the inverse square root of
         non-positive elements of :math:`d` as zeros.
+
+        Examples
+        --------
+        Prepare a symmetric positive definite covariance matrix ``A`` and a
+        data point ``x``.
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> rng = np.random.default_rng()
+        >>> n = 5
+        >>> A = np.diag(rng.random(n))
+        >>> x = rng.random(size=n)
+
+        Extract the diagonal from ``A`` and create the ``Covariance`` object.
+
+        >>> d = np.diag(A)
+        >>> cov = stats.Covariance.from_diagonal(d)
+
+        Compare the functionality of the ``Covariance`` object against a
+        reference implementations.
+
+        >>> res = cov.whiten(x)
+        >>> ref = np.diag(d**-0.5) @ x
+        >>> np.testing.assert_allclose(res, ref)
+        >>> res = cov.log_pdet
+        >>> ref = np.linalg.slogdet(A)[-1]
+        >>> np.testing.assert_allclose(res, ref)
```

Comment:
Not sure we need this part, but does not hurt and could help some. Same bellow.

Reply:
Which - the usage example?
I figure it helps clarify what's going on. Yeah, can't hurt.